test/test_main.py

```diff
@@ -465,3 +466,55 @@ def test_sysroot(tmp_path, capfd, verify_mod_cleanup, distutils_cache_cleanup):
         }
 
     assert data == expected
+
+
+def test_prefixed_sysroot(tmp_path, capfd, verify_mod_cleanup,
+                          distutils_cache_cleanup):
+    with open(tmp_path / "pyproject.toml", "w") as f:
+        f.write(ZIP_BACKEND_TOML.format(backend="sysroot_backend"))
+
+    stdlib_from = sysconfig.get_path("stdlib")
+    # Derive "BROOT", assuming regular Python prefix is /usr.
+    prefix_from = re.sub(r"/usr/.*", "", stdlib_from)
```

Comment:
This looks fragile. I think it'd be cleaner to strip path components starting from the rightmost `usr`.